shared/provision/set-public-name/container.tsx

```diff
@@ -1,49 +1,50 @@
 import * as ProvisionGen from '../../actions/provision-gen'
 import * as Constants from '../../constants/provision'
 import SetPublicName from '.'
-import {connect, withStateHandlers, compose, safeSubmit} from '../../util/container'
+import * as Container from '../../util/container'
 import {RouteProps} from '../../route-tree/render-route'
+import * as LoginGen from '../../actions/login-gen'
+import HiddenString from '../../util/hidden-string'
 
 type OwnProps = {
   deviceName: string
   onChange: (text: string) => void
 } & RouteProps<{}, {}>
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state: Container.TypedState) => ({
   _existingDevices: state.provision.existingDevices,
   error: state.provision.error.stringValue(),
+  loggedInAccounts: state.config.configuredAccounts
+    .filter(account => account.hasStoredSecret)
+    .map(account => account.username),
 })
 
-const mapDispatchToProps = (dispatch, ownProps: OwnProps) => ({
+const mapDispatchToProps = (dispatch: Container.TypedDispatch, ownProps: OwnProps) => ({
   _onSubmit: (name: string) => dispatch(ProvisionGen.createSubmitDeviceName({name})),
-  // TODO remove
-  onBack: () => {},
+  onLogIn: (username: string) => dispatch(LoginGen.createLogin({password: new HiddenString(''), username})),
 })
 
-const mergeProps = (stateProps, dispatchProps, ownProps) => {
-  const submitEnabled = !!(ownProps.deviceName.length >= 3 && ownProps.deviceName.length <= 64)
-  const onSubmit = submitEnabled ? () => dispatchProps._onSubmit(ownProps.deviceName) : null
-  return {
-    deviceName: ownProps.deviceName,
-    error: stateProps.error,
-    onBack: dispatchProps.onBack,
-    onChange: ownProps.onChange,
-    onSubmit,
-  }
-}
-
-export default compose(
-  withStateHandlers<any, any, any>(
+export default Container.compose(
+  Container.withStateHandlers<any, any, any>(
```

Comment:
ideally we'd remove this if we're in this code again. don't have to do this now as the PR is so big but in general we want to remove all recompose